aten/src/ATen/core/op_registration/op_registration.h

```diff
@@ -306,19 +278,19 @@ class CAFFE2_API RegisterOperators final {
       } else {
         return std::move(*this).kernel(
           std::move(dispatch_key),
-          KernelFunction::makeFromUnboxedOnlyFunction<FuncType, kernel_func>(),
+          KernelFunction::makeFromUnboxedOnlyRuntimeFunction(kernel_func),
           // TODO Do schema inference without relying on WrapKernelFunction
-          detail::FunctionSchemaInferer<typename detail::WrapKernelFunction<FuncType, kernel_func>::type>()()
+          detail::FunctionSchemaInferer<detail::WrapRuntimeKernelFunctor<guts::decay_t<FuncType>>>()()
         );
       }
     }
 
     // TODO Remove impl_unboxedOnlyCatchAllKernel once all of aten can generate boxed kernels
-    template<class FuncType, FuncType* kernel_func>
+    template<class FuncType>
     // enable_if: only enable it if FuncType is actually a function
-    guts::enable_if_t<guts::is_function_type<FuncType>::value, Options&&> impl_unboxedOnlyCatchAllKernel() && {
```

Comment:
this one is non-public and could be changed, but it would break XLA and other extension backends. You'd have to coordinate with changes there so they don't break. Also, I don't think you get any binary size improvements from changing this.